src/sys/inotify.rs

```diff
@@ -152,12 +152,12 @@ impl Inotify {
     ///
     /// For more information see, [inotify_add_watch(2)](https://man7.org/linux/man-pages/man2/inotify_add_watch.2.html).
     pub fn add_watch<P: ?Sized + NixPath>(
-        self,
+        &mut self,
```

Comment:
Why `&mut self` and not `&self`?

Reply:
Semantically, inotify_add_watch modifies the existing inotify instance, though this can not be reflected in the implementation of this binding.

[Man 2 inotify_add_watch](https://man7.org/linux/man-pages/man2/inotify_add_watch.2.html)

Reply:
I consider this is a case of "internal mutability". The correctness of the the method doesn't depend on it having exclusive access to the the structure, so we can remove the `mut`.

Reply:
Get it, this can also be applied to rm_watch and read_events